sympy/core/tests/test_count_ops.py

```diff
@@ -27,6 +28,9 @@ def count(val):
     assert count(Equivalent(x,y)) == 1
     assert count(ITE(x,y,z)) == 1
     assert count(ITE(True,x,y)) == 0
+    # not all args of Basic are Basic; the following
+    # might be 1 but should not fail to compute
+    assert count(OracleGate(1, lambda x: x == 2)) == 0
```

Comment:
I didn't realise what this PR was doing. I think that the OracleGate class should be fixed to have all Basic args.

We shouldn't try to accommodate for broken subclasses. They should be fixed. It is a good thing that various methods fail if a subclass breaks the invariant expected by the superclass because it makes it clear that there is a problem with that subclass.